docarray/array/doc_vec/doc_vec.py

```diff
@@ -116,64 +132,113 @@ def __init__(
             # from each document and put them in the corresponding column
             field_type = self.doc_type._get_field_type(field_name)
 
+            is_field_required = self.doc_type.__fields__[field_name].required
+
+            first_doc_is_none = getattr(docs[0], field_name) is None
+
+            def _check_none_field():
+
+                if is_field_required:
+                    if first_doc_is_none:
+                        raise ValueError(
+                            f'Field {field_name} is None for {docs[0]} even though it is required'
+                        )
+
+                if first_doc_is_none:
+                    for i, doc in enumerate(docs):
+                        if getattr(doc, field_name) is not None:
+                            raise ValueError(
+                                f'Field {field_name} is put to None for the first doc. This mean that all of the other docs should have this field set to None as well. This is not the case for {doc} at index {i}'
+                            )
+
+            def _check_field_not_none(field_name, doc):
+                if getattr(doc, field_name) is None:
+                    raise ValueError(
+                        f'Field {field_name} is None for {doc} even though it is not None for the first doc'
+                    )
+
             if is_tensor_union(field_type):
                 field_type = tensor_type
+
             if isinstance(field_type, type):
                 if tf_available and issubclass(field_type, TensorFlowTensor):
                     # tf.Tensor does not allow item assignment, therefore the
                     # optimized way
                     # of initializing an empty array and assigning values to it
                     # iteratively
                     # does not work here, therefore handle separately.
-                    tf_stack = []
-                    for i, doc in enumerate(docs):
-                        val = getattr(doc, field_name)
-                        if val is None:
-                            val = TensorFlowTensor(
-                                tensor_type.get_comp_backend().none_value()
-                            )
-                        tf_stack.append(val.tensor)
 
-                    stacked: tf.Tensor = tf.stack(tf_stack)
-                    tensor_columns[field_name] = TensorFlowTensor(stacked)
+                    if first_doc_is_none:
+                        _check_none_field()
+                        tensor_columns[field_name] = None
+                    else:
+                        tf_stack = []
+                        for i, doc in enumerate(docs):
```

Comment:
This loop seems really expensive to me, no? Iterating over all the data in python code? I think this is a problem with decent sized data